main.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/gptscript-ai/datasets/pkg/dataset"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		fmt.Println("missing argument")
+		os.Exit(1)
+	}
+
+	workspace := os.Getenv("GPTSCRIPT_WORKSPACE_DIR")
+	if workspace == "" {
+		fmt.Println("missing GPTSCRIPT_WORKSPACE_DIR")
+		os.Exit(1)
+	}
+
+	arg := os.Args[1]
+
+	var (
+		result string
+		err    error
+	)
+	switch arg {
+	case "info":
+		result, err = info(os.Getenv("ID"), workspace)
+	case "load_one":
+		result, err = loadOne(os.Getenv("ID"), os.Getenv("INDEX"), workspace)
+	case "load_range":
+		result, err = loadRange(os.Getenv("ID"), os.Getenv("START"), os.Getenv("END"), workspace)
+	case "load_all":
+		result, err = loadAll(os.Getenv("ID"), workspace)
+	}
+
```

Comment:
nit: check the error as close as possible to when it may be produced.
```suggestion
```

Reply:
no longer relevant after the reimplementation